src/shims/tls.rs

```diff
@@ -273,15 +278,36 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 StackPopCleanup::None { cleanup: true },
             )?;
 
-            // step until out of stackframes
-            this.run()?;
+            this.enable_thread(active_thread)?;
+            return Ok(());
+        }
+        this.machine.tls.last_dtor_key.remove(&active_thread);
+
+        Ok(())
+    }
+}
+
+impl<'mir, 'tcx: 'mir> EvalContextExt<'mir, 'tcx> for crate::MiriEvalContext<'mir, 'tcx> {}
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx> {
 
-            // Fetch next dtor after `key`.
-            dtor = match this.machine.tls.fetch_tls_dtor(Some(key), thread_id) {
-                dtor @ Some(_) => dtor,
-                // We ran each dtor once, start over from the beginning.
-                None => this.machine.tls.fetch_tls_dtor(None, thread_id),
-            };
+    /// Schedule an active thread's TLS destructor to run on the active thread.
+    /// Note that this function does not run the destructors itself, it just
+    /// schedules them one by one each time it is called.
+    ///
+    /// FIXME: we do not support yet deallocation of thread local statics.
```

Comment:
Given that this is the library TLS file, this seems like an odd place for the comment?

Please also open an issue for this, and reference it from here.